src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,1436 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_ORC_PUP_JUST_SUMMONED:
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                        Talk(TALK_ORC_PUP_SUMMONED, player);
+                    break;
+
+                case EVENT_ORC_PUP_TALK:
+                {
+                    if (me->GetVehicleKit()->IsVehicleInUse())
+                    {
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                            Talk(TALK_ORC_PUP_WORN_OFF, player);
+                    }
+                    _events.ScheduleEvent(EVENT_ORC_PUP_TALK, 30s, 75s);
+                    break;
+                }
+
+                case EVENT_ORC_PUP_REMOVE_PROTECTION:
+                    _isJustSummoned = false;
+                    break;
+
+                case EVENT_ORC_PUP_DELIVER_CRATES:
+                    DoCastSelf(SPELL_EJECT_ALL_PASSENGERS);
+                    me->SetStandState(UNIT_STAND_STATE_DEAD);
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+    bool _isJustSummoned;
+};
+
+enum OrcCrate
+{
+    NPC_ORC_SEA_PUP                             = 44914
+};
+
+// 44915 - Orc Crate
+struct npc_silverpine_orc_crate : public ScriptedAI
+{
+    npc_silverpine_orc_crate(Creature* creature) : ScriptedAI(creature) { }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (summoner->GetEntry() == NPC_ORC_SEA_PUP)
+        {
+            me->SetUnitFlag(UNIT_FLAG_UNINTERACTIBLE);
+
+            me->EnterVehicle(summoner->ToCreature());
+        }
+    }
+};
+
+enum PickUpOrcCrate
+{
+    SPELL_KILL_CREDIT_SEA_DOG_CRATE             = 83843
+};
+
+// 83838 - Pick Up Orc Crate
+class spell_silverpine_pick_up_orc_crate : public SpellScript
+{
+    PrepareSpellScript(spell_silverpine_pick_up_orc_crate);
+
+    bool Validate(SpellInfo const* /*spellInfi*/) override
+    {
+        return ValidateSpellInfo(
+        {
+            SPELL_SUMMON_ORC_CRATE,
+            SPELL_KILL_CREDIT_SEA_DOG_CRATE
+        });
+    }
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        GetHitUnit()->CastSpell(nullptr, SPELL_SUMMON_ORC_CRATE);
+        GetHitUnit()->CastSpell(nullptr, SPELL_KILL_CREDIT_SEA_DOG_CRATE);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_silverpine_pick_up_orc_crate::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
+    }
+};
+
+// 83840 - Despawn All Summons
+class spell_silverpine_despawn_all_summons_rear_guard : public SpellScript
+{
+    PrepareSpellScript(spell_silverpine_despawn_all_summons_rear_guard);
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
```

Comment:
caster can be null

Reply:
Feedback applied, I'm removing the spell from here since it was added in the last PR.